syft/execution/role_assignments.py

```diff
@@ -10,13 +16,20 @@ class RoleAssignments:
     which other parties are participating and communicate with them if needed.
     """
 
-    def __init__(self, role_ids):
+    def __init__(self, role_ids: list = [], assignments: dict = {}):
         """
         Args:
             role_ids: an iterable containing values that indentify the roles of
                 the Protocol to which the RoleAssignments is associated.
         """
-        self.assignments = {role_id: [] for role_id in role_ids}
+        if assignments:
+            self.assignments = assignments
+        elif role_ids:
+            self.assignments = {role_id: [] for role_id in role_ids}
+        else:
+            raise ValueError(
+                "You need to provide role_ids or assignments in RoleAssignments' constructor"
```

Comment:
I think this is a typo, should be RoleAssignament's.

Reply:
The class name is plural, so there is no typo, I think (English is not my native language so I may be wrong though ^^)

Reply:
Bizarrely, this is the correct punctuation in English. It's a shit language and I make no excuses.